### ISSUE TITLE
Throughput metric is not taking into account the number of processes

### ISSUE THREAD

**Issue by gwenzek** [pytorch/torcheval#156]
### 🐛 Describe the bug

I'm using `torcheval.metrics.Throughput` to compute a number of "tokens per second" for my training loop.
To avoid synchronization overhead I don't sync the metric every time I log it, but only once in a while.
In this examples I'm using a 8 process job with 8 GPUs

```py

metrics["tps"] = torcheval.metrics.Throughput(device=device)
...

metrics["tps"].update(tgt_num_tokens, elapsed_time_sec=state.timer.interval_time_seconds)
... 

should_log = (step % freq == 0)
should_sync = step % self.sync_frequency == 0
if should_log:
    if should_sync:
        val = torcheval.metrics.toolkit.sync_and_compute(metrics["tps"]).item()
    else:
        val = metrics["tps"].compute().item()
    self.log_metric({"train/tps": val})
```

![image](https://github.com/pytorch/torcheval/assets/5920036/8ce68eeb-fba1-492c-98e5-ffc8ab41f981)

This creates big spikes in the log graph, because throughput is not averaging over the number of workers, so when I sync I get x8 TPS. 

Is this the expected behavior ? I would find it less surprising if sync was returning the average throughput, or if non-sync was returning an estimated global throughput.

### Versions

```
[conda] torch                     2.0.0+cu117              pypi_0    pypi
[conda] torchaudio                2.0.1+cu117              pypi_0    pypi
[conda] torcheval                 0.0.5                    pypi_0    pypi
[conda] torchsnapshot             0.1.0                    pypi_0    pypi
[conda] torchsnapshot-nightly     2022.11.28               pypi_0    pypi
[conda] torchtnt                  0.0.7                    pypi_0    pypi
[conda] torchx                    0.5.0                    pypi_0    pypi
[conda] triton                    2.0.0                    pypi_0    pypi
```

**Comment by ananthsub** [comment]
Hi @gwenzek, calling compute() on its own does jot take into account the number of processes. If not using the toolkits sync_and_compute, you'd need to manually multiply by the number of processes running. 

You can see the example here: https://github.com/pytorch/torcheval/blob/main/examples/distributed_example.py

In particular, lines 146-150 address how to compute an "estimated" throughput value, which assumes that all processes are roughly equal.

**Comment by bobakfb** [comment]
closing since it seems this was not a bug
